### PR TITLE
Winbuild

### PR DESCRIPTION
**Pull request by nicolaasuni**
[PE-186]
* With this version is possible to cross-compile the Windows version in Linux using mingw and wine.
* The dockerbuild.sh script now builds and test bouth Linux and Windows versions for 32 and 64 bit.
* For cross compatibility (suppress warning on Windows), in "big.c" the "long long unsigned int" has been replaced with "uint64_t" and printed using the macro "PRIxMAX" defined in inttypes.h.
* The Travis CI configuration has been updated to build and test also the windows versions.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,15 +31,21 @@ addons:
     - libffi-dev
     - doxygen
     - doxygen-latex
+    - mingw-w64
```
Comment (nicolaasuni) [inline .travis.yml]: now travis build also the windows versions

File: doc/AMCL.dox [hunk 2]
```diff
@@ -140,9 +140,9 @@ export PYTHONPATH=/usr/lib/python2.7/dist-packages
 <p>Start a command prompt as an administrator</p>
 <pre><code>git clone https://github.com/miracl/milagro-crypto-c
 cd milagro-crypto-c
-mkdir -p target/build
-cd target/build
-cmake -G "MinGW Makefiles" ../..
+mkdir -p target\\build
```
Comment (nicolaasuni) [inline doc/AMCL.dox]: this was breaking the documentation as the backslashes weren't escaped

File: include/amcl.h.in [hunk 3]
```diff
@@ -44,6 +44,7 @@ under the License.
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <inttypes.h>
```
Comment (nicolaasuni) [inline include/amcl.h.in]: this is required in order to use the macro PRIxMAX in big.c

File: resources/DockerDev/Dockerfile [hunk 4]
```diff
@@ -50,13 +50,20 @@ ENV DISPLAY :0
 ENV GOPATH=/root/GO
 ENV PATH=$GOPATH/bin:$PATH
 
+# add i386 architecture
+RUN dpkg --add-architecture i386
```
Comment (nicolaasuni) [inline resources/DockerDev/Dockerfile]: the docker development image now contains the packages for windows cross-compilation

File: src/big.c [hunk 5]
```diff
@@ -119,8 +119,8 @@ void BIG_rawoutput(BIG a)
     int i;
     printf("(");
     for (i=0; i<NLEN-1; i++)
-        printf("%llx,",(long long unsigned int) a[i]);
-    printf("%llx)",(long long unsigned int) a[NLEN-1]);
+        printf("%"PRIxMAX",",(uint64_t) a[i]);
```
Comment (nicolaasuni) [inline src/big.c]: the "llx" modifier is not supported in Windows.

File: test/CMakeLists.txt [hunk 6]
```diff
@@ -4,10 +4,15 @@
 link_directories (${PROJECT_BINARY_DIR}/src)
 include_directories (${CMAKE_CURRENT_BINARY_DIR} ${PROJECT_SOURCE_DIR}/include ${PROJECT_BINARY_DIR}/include )
 
+set(EXE_FILE_EXTENSION "")
+if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  # set the executable file extension for windows tests
+  set(EXE_FILE_EXTENSION ".exe")
```
Comment (nicolaasuni) [inline test/CMakeLists.txt]: when cross-compiling Windows in Linux it is necessary to specify the executable extension.

**Comment by AlessandroBudroni** [comment]
Great job!

**Pull request merged**